graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/processor/QueryMappingModel.java

```diff
@@ -72,33 +69,52 @@ public EntityMappingModel getEntity() {
       String parentTypeName,
       Map<String, EntityMappingModel> entities,
       ProcessingContext context) {
+    System.out.println(
+        "QueryMappingModel.build, query: "
+            + query
+            + " , parentTypeName: "
+            + parentTypeName
+            + " , entities: "
+            + entities
+            + "context:"
+            + context);
 
     Type<?> returnType = query.getType();
-    String entityName = (returnType instanceof TypeName) ? ((TypeName) returnType).getName() : null;
+    boolean isListType = returnType instanceof ListType;
+    String entityName = null;
```

Comment:
I am not sure if this is the correct approach. 
Let's assume that we have a given graphql schema and query:
```
"type Foo {\n"
                    + "  pk1: Int! @cql_column(partitionKey: true)\n"
                    + "  pk2: Int! @cql_column(partitionKey: true)\n"
                    + "  cc1: Int! @cql_column(clusteringOrder: ASC)\n"
                    + "  cc2: Int! @cql_column(clusteringOrder: DESC)\n"
                    + "}"
                    + "type Query { foo1(pk1: Int!, pk2: Int!, cc1: Int!): [Foo] }"
```
The map of entities will contain only `Foo` entity definition. I wonder If we should add a list of type when building a map of entities?
In the `MappingModel` we have this logic that builds the `EntityMappingModel` for `Foo` :
```
    for (ObjectTypeDefinition type : registry.getTypes(ObjectTypeDefinition.class)) {
      //...
      new EntityMappingModelBuilder(type, context)
          .build()
          .ifPresent(entity -> entitiesBuilder.put(type.getName(), entity));
    }
```
maybe for all non `query` `ObjectTypeDefinition type` we should build a normal Foo type but also a list of foo (since we will support that now)?
If yes, then we need some convention of how those list of foo will be stored in the entities map (the key is a string).
Maybe we should create `list_of_Foo`?
So the logic will be:
```

    for (ObjectTypeDefinition type : registry.getTypes(ObjectTypeDefinition.class)) {
      if (typesToIgnore.contains(type)) {
        continue;
      }
      new EntityMappingModelBuilder(type, context)
          .build()
          .ifPresent(entity -> entitiesBuilder.put(type.getName(), entity));
   // for every entity build a corresponding list[entity]
   new EntityMappingModelBuilder(type, context) // new mapping? or pass the info isList to an existing Builder?
          .build()
          .ifPresent(entity -> entitiesBuilder.put("list_of_" + type.getName(), entity));
    }
```
However, in that case, we would need to create a variant of `EntityMappingModel` that contains a `List<EntityMappingModel>`. 
Or maybe this `entity -> list[entity]` mapping should be handled elsewhere and we don't need an additional entry in the entities map - @olim7t wdyt?

Reply:
I think the simplest approach is to encode this information in a separate field: `QueryMappingModel.returnsList()`. Then in `QueryFetcher` it will be pretty easy to process the result set accordingly: if `returnsList()`, read all the rows and return a list of maps.